server/src/main/java/org/opensearch/transport/TransportHandshaker.java

```diff
@@ -131,8 +131,9 @@ void handleHandshake(TransportChannel channel, long requestId, StreamInput strea
         // 1. if remote node is 7.x, then StreamInput version would be 6.8.0
         // 2. if remote node is 6.8 then it would be 5.6.0
         // 3. if remote node is OpenSearch 1.x then it would be 6.7.99
-        if ((this.version.onOrAfter(Version.V_1_0_0) && this.version.before(Version.V_2_0_0))
+        if ((this.version.onOrAfter(Version.V_1_0_0) && this.version.before(Version.fromId(3000099 ^ Version.MASK)))
```

Comment:
We can bake it as a method, yes. Not a constant, though, because of the crazy dependency between [GlobalBuildInfoPlugin](https://github.com/opensearch-project/OpenSearch/blob/0f31d62cd9aac1c9c2ddddbe30fce8523a997a0c/buildSrc/src/main/java/org/opensearch/gradle/info/GlobalBuildInfoPlugin.java#L76). This will also be removed in 3.0.0 so its time here on earth is limited.

Reply:
I'll bake it as a private static specific to TransportHandshaker class.

Reply:
Sounds good, thanks @nknize !